### ISSUE TITLE
`previous` field in `NamedResourceApiList` is incorrect type

### ISSUE THREAD

**Issue by milomarten** [SirSkaro/pokeapi-reactor#8]
NamedApiResourceList seems to have the incorrect type for the `previous` field. PokeAPI actually returns a string for this field, being the URL of the previous page of results:
```json
{
  count:1302
  next:"https://pokeapi.co/api/v2/pokemon?offset=11&limit=10"
  previous:"https://pokeapi.co/api/v2/pokemon?offset=0&limit=1"
  results: ...
}
```

As such, program throws an exception on deserialization.
On the first page of results (offset == 0), `previous` is null, and deserialization happens successfully.

To Replicate:
```java
@Component
public class PokemonLooper implements ApplicationRunner {
    @Autowired
    private PokeApiClient client;

    @Override
    public void run(ApplicationArguments args) throws Exception {
        client.getResource(Pokemon.class, new PageQuery(10, 1))
                .flatMapIterable(l -> l.getResults())
                .subscribe(n -> System.out.println(n.getName()));
    }
}
```
- Expected behavior:  Pokemon numbers 2 to 11 are printed to console.
- Actual behavior: `JSON decoding error: Cannot deserialize value of type `java.lang.Boolean` from String "https://pokeapi.co/api/v2/pokemon?offset=0&limit=1": only "true" or "false" recognized`.

**Comment by SirSkaro** [comment]
Thanks for reporting! I'm a bit busy with school + personal things right now, so I probably won't get around to taking a look at this for a bit. When I do I'll use your replication (thank you for providing that) and push a fix.

That said, if anyone else wants to contribute before I have a chance to, please feel free.

**Comment by SirSkaro** [comment]
Fixed by #10 - thank you @aaronhunter1088 !
The bug fix has been published in the new 1.0.7 release. Happy coding!
